go/vt/vtgate/tx_conn.go

```diff
@@ -201,25 +201,51 @@ func (txc *TxConn) commit2PC(ctx context.Context, session *SafeSession) error {
 		return err
 	}
 
+	// retrieve the caller ID to execute the testing flow
+	callerID := callerid.EffectiveCallerIDFromContext(ctx)
+
+	// Test code to simulate a failure after RM prepare
+	if failNow, err := checkTestFailure(callerID, "TRCreated_FailNow", nil); failNow {
```

Comment:
I'm confused. Isn't this test code? What is it doing here?

Reply:
We will add the `go build` logic to include test and non-test functions. There is already a task to make the changes on the commit flow. That will be added as part of the PR.

Reply:
Done! In the latest commits! 🚀